src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,346 @@
+    }, []);
+
+    const indicatorTransitionInterpolate = useMemo(() => {
+        const countItems = validChildren.length;
+        if (countItems < 2 || !tabItemPositions.current.length) {
+            return 0;
+        }
+        const inputRange = Array.from(Array(countItems).keys());
+        const outputRange = tabItemPositions.current.map(({ position }) => position);
+
+        return animationRef.current.interpolate({
+            inputRange,
+            outputRange,
+        });
+        // to make useMemo in sync with the ref
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [animationRef, variant, validChildren, tabItemPositions.current.length]);
+
+    const WIDTH = useMemo(() => {
+        return variant === 'primary'
+            ? tabItemPositions.current[valueIndex]?.contentWidth
+            : tabItemPositions.current[valueIndex]?.width;
+        // to make useMemo in sync with the ref
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [value, variant, tabItemPositions.current.length]);
+
+    const { containerStyle, itemsContainerStyle, dividerStyle, indicatorStyle } = useMemo(() => {
+        const { indicator, itemsContainer, divider, activeColor, ...restStyle } = componentStyles;
+
+        return {
+            containerStyle: restStyle,
+            itemsContainerStyle: itemsContainer,
+            dividerStyle: divider,
+            indicatorStyle: [
+                indicator,
+                {
+                    backgroundColor: activeColor,
+                    transform: [
+                        {
+                            translateX: indicatorTransitionInterpolate,
+                        },
+                    ],
+                    width: WIDTH,
+                },
+            ],
+        };
+    }, [WIDTH, componentStyles, indicatorTransitionInterpolate]);
+
+    const Container = scrollable ? ScrollView : View;
+    const containerProps = scrollable
+        ? {
+              horizontal: true,
+              ref: scrollViewRef,
+              onScroll: onScrollHandler,
+              showsHorizontalScrollIndicator: false,
+              style: itemsContainerStyle,
+          }
+        : {};
+
+    const onLayout = useCallback(({ nativeEvent: { layout } }: LayoutChangeEvent) => {
+        setTabContainerWidth(layout.width);
+    }, []);
+
+    const onLayoutItem = useCallback(
+        (event: LayoutChangeEvent, index: number) => {
+            const { width } = event.nativeEvent.layout;
+
+            const previousItemsWidth = tabItemPositions.current.reduce((acc, item, i) => {
+                if (index <= i) return acc;
+
+                acc += item.width || 0;
+
+                return acc;
+            }, 0);
+
+            const currentItemPosition = tabItemPositions.current[index];
+
+            tabItemPositions.current[index] = {
+                ...currentItemPosition,
+                position:
+                    variant === 'primary'
```

Comment:
this value can be derived where you need it. storing the value like this prevents you from being able to switch the variant.